src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -778,7 +776,27 @@ public void CompileMethod(MethodWithGCInfo methodCodeNodeNeedingCode, Logger log
 
                     // Harvest the method being compiled for the purpose of populating the type resolver
                     var resolver = _compilation.NodeFactory.Resolver;
-                    resolver.AddModuleTokenForMethod(MethodBeingCompiled, new ModuleToken(ecmaMethod.Module, ecmaMethod.Handle));
+                    if (_compilation.CompilationModuleGroup.VersionsWithModule(ecmaMethod.Module))
+                    {
+                        resolver.AddModuleTokenForMethod(MethodBeingCompiled, new ModuleToken(ecmaMethod.Module, ecmaMethod.Handle));
+                    }
+                    else
+                    {
+                        // Cross module optimization scenario. If we aren't able to come up with a token for the method being compiled, then force it to be
+                        // computed.
```

Comment:
assume the computation is deterministic based on the methodName, returnValue etc?

Reply:
Yes. Computation is deterministic as setting the `_ilBodiesNeeded` variable will defer work until single threaded, and then will sort the list and fill stuff in deterministically.

Reply:
sounds good.